src/GeneralStateTestsFiller/stReturnDataTest/clearReturnBufferFiller.yml

```diff
@@ -0,0 +1,518 @@
+             mstore8(0x57, 0x3D)
+         
+             //   0x58 PUSH1 1
+             mstore8(0x58, 0x60)
+             mstore8(0x59, 0x01)
+
+             //   0x5A SSTORE
+             mstore8(0x5A, 0x55)
+
+             //   0x5B STOP
+             mstore8(0x5B, 0x00)
+
+             // Return the length
+             retVal := 0x5C
+              
+           }   // function makeConstructor
+
+
+           // Put the constructor code at 0x00-length, and return that length - but this
+           // constructor fails with an illegal opcode
+           function makeFailConstructor(addr, len) -> retVal {
+              let endOfGood := makeConstructor(addr, len)
+              
+              // Replace the STOP with 0xFE, which is the designated bad opcode
+              mstore8(sub(endOfGood, 1), 0xFE)
+              retVal := endOfGood
+           }
+
+
+           switch operation
+           case 0xF0 {  // CREATE
+              codeLen := makeConstructor(bufferFrom, bufLen)
+              pop(create(0, 0, codeLen))
+           }
+           case 0x11F0 {  // CREATE, which fails
+              codeLen := makeFailConstructor(bufferFrom, bufLen)
+              pop(create(0, 0, codeLen))
+           }
+           case 0xF5 {  // CREATE2
+              codeLen := makeConstructor(bufferFrom, bufLen)
+              pop(create2(0, 0, codeLen, 0x5a17))
+           }
+           case 0x11F5 {  // CREATE2, which fails
+              codeLen := makeFailConstructor(bufferFrom, bufLen)
+              pop(create2(0, 0, codeLen, 0x5a17))
+           }
+           case 0xF1 { // CALL
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(call(gas(), 0x00ca11ed, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11F1 { // CALL, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(call(gas(), 0xbad0ca11, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0xF2 { // CALLCODE
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(callcode(gas(), 0x00ca11ed, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11F2 { // CALLCODE, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(callcode(gas(), 0xbad0ca11, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0xF4 { // DELEGATECALL
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(delegatecall(gas(), 0x00ca11ed, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11F4 { // DELEGATECALL, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(delegatecall(gas(), 0xbad0ca11, 0, 0x40, 0x100, 0x20))
+           }
+           case 0xFA { // STATICCALL
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(staticcall(gas(), 0x00ca11ed, 0, 0x40, 0x100, 0x20))
```

Comment:
in case of static call you need to work with the memory to verify conditions
otherwise the sstore operation in subcall will oog

Reply:
Good point. I created a new contract, 0x57A700ca11ed, with sstore's.